fix: bridge and swap actions context params generation

ShahSujal opened the pull request:
# Relates to:
issue number #1523 

# Risks
Medium: Changes on two files created compose actions & regenerate the details object

# Background

## What does this PR do?
This fix ensures that the bridge and swap actions are properly populated with the necessary details required for the bridgeAction and SwapAction handlers. As a result:

Bridge operations will now process the toAddress, token information, and other required parameters without error.
Swap operations will execute successfully with all required data passed to the handler.

shakkernerd committed:
fix: move no response handler up

madjin committed:
add greenpill notes

shakkernerd committed:
Merge pull request #1469 from todorkolev/main

fix: cleaner interaction prompts in the Twitter plugin

ag-wnl committed:
feat: ability to fetch timeline resticted to accounts followed

shakkernerd committed:
Merge branch 'develop' into tcm-twitter-duplicate-post

shakkernerd committed:
Merge pull request #1472 from elizaOS/tcm-twitter-duplicate-post

fix: duplicate twitter post

shakkernerd committed:
Merge branch 'develop' into docs_update

shakkernerd committed:
Merge pull request #1476 from elizaOS/docs_update

chore: Docs update

mcp97 committed:
Merge branch 'develop' into fetch-following-timeline

ag-wnl committed:
chore: do not consider self tweets when evaluating actions

shakkernerd committed:
Merge pull request #1475 from ag-wnl/fetch-following-timeline

Fetch timeline for followed accounts via Twitter client methods

shakkernerd committed:
Merge branch 'develop' into ignore-self-tweets-for-action

shakkernerd committed:
Merge pull request #1477 from ag-wnl/ignore-self-tweets-for-action

chore: Do not consider self tweets when evaluating actions

shakkernerd committed:
Merge branch 'develop' into 1223-fix-todos

shakkernerd committed:
Merge pull request #1408 from harperaa/1223-fix-todos

fix: resolve TODO to make model more dynamic and generic

shakkernerd committed:
feat: response message memory composition

shakkernerd committed:
fix: move process actions up and remove unuse variable

shakkernerd committed:
feat: update recent messages state

azep-ninja committed:
Merge branch 'develop' into feature/add-gitbook-provider

shakkernerd committed:
chore: rework message ids in memory

shakkernerd committed:
Merge branch 'develop' into enhance/client-direct

0xPBIT committed:
images in chat

shakkernerd committed:
Merge pull request #1479 from elizaOS/enhance/client-direct

feat: Enhance client direct

0xPBIT committed:
fix lockfile

swizzmagik committed:
Merge remote-tracking branch 'upstream/Main'

swizzmagik committed:
feat: improve chat formatting and add support for line breaks

swizzmagik committed:
chore: remove .env

swizzmagik committed:
chore: remove .env

shakkernerd committed:
chore: re-order components imports

shakkernerd committed:
chore: re-order useMutation import

shakkernerd committed:
Merge branch 'develop' into feat-improve-chat-formatting-line-breaks

shakkernerd committed:
Merge pull request #1483 from swizzmagik/feat-improve-chat-formatting-line-breaks

feat: improve chat formatting line breaks

mcp97 committed:
Merge branch 'develop' into images-in-chat-client

mcp97 committed:
Update Chat.tsx

shakkernerd committed:
fix: remove File import

shakkernerd committed:
Merge pull request #1481 from 0xPBIT/images-in-chat-client

feat: add image features to react chat client

Ed-Marcavage committed:
change repo owner to elizaOS

Ed-Marcavage committed:
Merge branch 'develop' into agentic-JSDoc

shakkernerd committed:
Merge branch 'develop' into twitter-convo

shakkernerd committed:
chore: update agent-twitter-client version to v0.0.17

shakkernerd committed:
chore: pnpm lock file

Ed-Marcavage committed:
Move to /scripts dir

Ed-Marcavage committed:
Merge branch 'agentic-JSDoc' of github.com:AudixAI/eliza into agentic-JSDoc

Ed-Marcavage committed:
change repo owner to elizaOS

Shane0xM committed:
fix: add required incremental option and remove invalid typescript config

Ed-Marcavage committed:
Merge branch 'develop' into agentic-JSDoc

Shane0xM committed:
Merge branch 'main' into fix/typescript-config

mcp97 committed:
Merge branch 'develop' into fix/typescript-config

nulLeeKH committed:
add CODE_OF_CONDUCT.md

with default template

madjin committed:
Merge pull request #1343 from AudixAI/agentic-JSDoc

feat: Add agentic JSDoc generation

nulLeeKH committed:
Merge branch 'develop' into patch-1

Shane0xM committed:
fix: improve client type identification in runtime initialization - Add determineClientType helper function to reliably identify client types - Replace client.name usage with robust type detection - Maintain backward compatibility with existing clients - Add debug logging for client initialization

nulLeeKH committed:
update contact in CODE_OF_CONDUCT.md

Shane0xM committed:
test: add Jest testing setup for client type identification - Add Jest and ts-jest dependencies - Configure Jest for TypeScript - Add test script to package.json

Shane0xM committed:
test: add tests for client type identification - Test explicit type property handling - Test constructor name fallback - Test unique identifier generation for plain objects

azep-ninja committed:
Merge branch 'develop' into feature/add-gitbook-provider

qwdsds committed:
chore: fix typos

Shane0xM committed:
Merge branch 'main' into fix/client-type-identification

tomguluson92 committed:
Update index.ts

tomguluson92 committed:
Update attachTerms.ts

tomguluson92 committed:
Update getAvailableLicenses.ts

tomguluson92 committed:
Update getIPDetails.ts

tomguluson92 committed:
Update licenseIP.ts

tomguluson92 committed:
Update registerIP.ts

tomguluson92 committed:
Update wallet.ts

mdqst committed:
docs: Fixed a small syntax issue in the ModelClass Update fine-tuning.md

chandiniv1 committed:
feat: add readme for websearch plugin

chandiniv1 committed:
feat: add readme for ton plugin

e-fu committed:
fix: improve Twitter client dry run mode and configuration logging

- Added proper boolean parsing for dry run configuration
- Improved configuration logging on startup
- Prevented tweet generation and action loops in dry run mode
- Added more descriptive dry run messages
- Fixed duplicate loop execution

pythonberg1997 committed:
fix: swap and bridge actions of plugin-evm

tcm390 committed:
Merge branch 'develop' into tcm-long-tweet

lalalune committed:
Merge pull request #1496 from chandiniv1/chandini/plugin-ton-docs

feat: add readme for ton plugin

lalalune committed:
Merge pull request #1494 from chandiniv1/chandini/websearch-plugin-docs

feat: add readme for websearch plugin

lalalune committed:
Merge pull request #1489 from qwdsds/chore/typos

chore: fix typos

lalalune committed:
Merge pull request #1493 from mdqst/patch-2

docs: Fixed a small syntax issue in the ModelClass Update fine-tuning.md

lalalune committed:
Merge pull request #1487 from nulLeeKH/patch-1

add CODE_OF_CONDUCT.md

lalalune committed:
Merge pull request #1492 from tomguluson92/tomguluson92-patch-2

fix: remove `type` when import from `elizaos`

shakkernerd committed:
chore: remove console.log and set debug logs

shakkernerd committed:
fix: move dry run return to tweet reply

azep-ninja committed:
Merge branch 'develop' into feature/add-gitbook-provider

janeyJo committed:
bad links in eliza-in-tee.md

shakkernerd committed:
Merge branch 'develop' into pr-1498

shakkernerd committed:
feat: extend parseBooleanFromText function with additional boolean values

- Added support for "ENABLE" and "DISABLE" as affirmative and negative cases respectively.
- Ensured consistent handling of null, undefined, and unrecognized inputs.

shakkernerd committed:
Merge pull request #1501 from elizaOS/enhance/core

feat: extend parseBooleanFromText function with additional boolean values

shakkernerd committed:
Merge branch 'develop' into patch-1

shakkernerd committed:
Merge pull request #1500 from janeyJo/patch-1

docs: bad links in eliza-in-tee.md

tcm390 committed:
resolve conflict

tcm390 committed:
resolve conflict

lalalune committed:
Merge branch 'main' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

shakkernerd committed:
Merge branch 'develop' into pr-1490

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge branch 'develop' of github.com:elizaOS/eliza into develop

shakkernerd committed:
Merge branch 'develop' into tcm-long-tweet

Rudrakc committed:
Init commit for the fere plugin

shakkernerd committed:
Merge pull request #1339 from elizaOS/tcm-long-tweet

feat: handle long tweet

timolegros committed:
Clear `/cache/` in `clean.sh` script

odilitime committed:
isFalsish() for parsing environment in a standard way

odilitime committed:
isFalsish() for parsing environment in a standard way

timolegros committed:
Merge branch 'main' into 1506--clean-cache

odilitime committed:
roll isFalsish into parseBooleanFromText

odilitime committed:
roll isFalsish into parseBooleanFromText

Dhaiwat10 committed:
add fuel plugin

shakkernerd committed:
Merge branch 'develop' into fixes

shakkernerd committed:
Merge branch 'develop' into 1506--clean-cache

Dhaiwat10 committed:
add docs

shakkernerd committed:
fix: remove EmbeddingProvider namespace

shakkernerd committed:
fix: add error variable to log

shakkernerd committed:
fix: remove double Action import

shakkernerd committed:
fix: remove unused imports

Dhaiwat10 committed:
fix conflicts in lockfile

shakkernerd committed:
chore: comment out unused imports and interface

shakkernerd committed:
Merge pull request #1513 from elizaOS/fixes

chore: general code fixes/clean up

Dhaiwat10 committed:
reset dev.sh

Dhaiwat10 committed:
Merge branch 'develop' into 1510--add-fuel-plugin

mcp97 committed:
Merge branch 'develop' into 1506--clean-cache

odilitime committed:
lint: remove unused vars

odilitime committed:
lint: remove unused error variable

odilitime committed:
lint: fix EmbeddingProvider already defined error (via claude)

odilitime committed:
empty is already mapped to null

odilitime committed:
empty is already mapped to null

odilitime committed:
move all env parsing here

odilitime committed:
move all env parsing here

odilitime committed:
add twitterConfig cstr param and prefer it over getSetting

odilitime committed:
add twitterConfig cstr param and prefer it over getSetting

shakkernerd committed:
chore: add required fuel env

odilitime committed:
prefer this.client.twitterConfig over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSetting

shakkernerd committed:
chore: remove .env.example in package dir

odilitime committed:
prefer client.twitterConfig over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSetting

odilitime committed:
prefer this.client.twitterConfig over getSettings

odilitime committed:
prefer this.client.twitterConfig over getSettings

odilitime committed:
pass twitterConfig as 2nd parameter to cstr, only start search once

odilitime committed:
pass twitterConfig as 2nd parameter to cstr, only start search once

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into test-eliza

mcp97 committed:
Merge branch 'develop' into fix/typescript-config

mcp97 committed:
Merge pull request #1485 from ShaneOxM/fix/typescript-config

fix: add required incremental option and remove invalid typescript configuration

mcp97 committed:
Merge branch 'develop' into 1506--clean-cache

odilitime committed:
fix TWITTER_POLL_INTERVAL scale

odilitime committed:
fix TWITTER_POLL_INTERVAL scale

odilitime committed:
Merge branch 'develop' into test-eliza

odilitime committed:
make MAX_TWEET_LENGTH integer since that's how it's consomed/compared

odilitime committed:
make MAX_TWEET_LENGTH integer since that's how it's consomed/compared

odilitime committed:
Merge branch 'test-eliza' of https://github.com/odilitime/eliza into test-eliza

mcp97 committed:
Merge pull request #1508 from timolegros/1506--clean-cache

Clear `/cache/` in `clean.sh` script

shakkernerd committed:
Revert "Clear `/cache/` in `clean.sh` script"

shakkernerd committed:
Merge pull request #1515 from elizaOS/revert-1508-1506--clean-cache

chore: Revert "Clear `/cache/` in `clean.sh` script"

shakkernerd committed:
chore: remove cache in core

shakkernerd committed:
Merge pull request #1516 from elizaOS/fix/remove_cache

chore: remove cache in core

shakkernerd committed:
Merge branch 'develop' into add-fereProPlugin

shakkernerd committed:
chore: remove package-lock file

shakkernerd committed:
chore: remove empty README file

shakkernerd committed:
feat: add eslint support

shakkernerd committed:
chore: remove unused imports

shakkernerd committed:
chore: remove typescript from package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: add npmignore file

shakkernerd committed:
Merge branch 'develop' into test-eliza

odilitime committed:
Merge branch 'develop' into patch-1

shakkernerd committed:
revert: packages/client-github/src/index.ts to match develop

odilitime committed:
update org

odilitime committed:
Merge pull request #1312 from y4my4my4m/patch-1

fix: Update speech.ts

odilitime committed:
Merge branch 'develop' into fix

odilitime committed:
Merge pull request #1456 from pythonberg1997/fix

fix: swap and bridge actions of plugin-evm

odilitime committed:
Merge branch 'develop' into test-eliza

oxSaturn committed:
fix: handle long tweet in utils

imtms committed:
feat: use OPENAI_API_URL from env to support custom OpenAI API endpoint

imtms committed:
feat: use OPENAI_API_URL from env to support custom OpenAI API endpoint

odilitime committed:
Merge branch 'develop' into fix/long-tweet

ShahSujal committed:
fix: bridge and swap actions context params generation

shakkernerd committed:
Merge branch 'develop' into pr-1514

shakkernerd committed:
chore: revert file to match develop branch

shakkernerd committed:
chore: revert file to match develop branch

shakkernerd committed:
chore: revert file to match develop branch

shakkernerd committed:
chore: revert file to match develop branch

Titan-Node committed:
add Livepeer configuration to .env.example

Titan-Node committed:
add Livepeer gateway URL to agent configuration

Titan-Node committed:
add Livepeer model configuration

Titan-Node committed:
add Livepeer model and provider name to core types

shakkernerd committed:
Merge branch 'develop' into test-eliza

Titan-Node committed:
Included image generation action to include validation for Livepeer gateway URL

Titan-Node committed:
Implement Livepeer image generation logic with URL validation

shakkernerd committed:
Merge pull request #1514 from odilitime/test-eliza

fix: client-twitter lowerCase bug and environment clean up (+lint fixes, and TWITTER_SEARCH_ENABLE double start fix)

tsubasakong committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge branch 'develop' into develop

samarth30 committed:
fix: fixed dependecny issue and resolved merge conflicts

shakkernerd committed:
Merge pull request #1522 from imtms/develop

feat: use OPENAI_API_URL from env to support custom OpenAI API endpoint

odilitime committed:
lint: remove unused error variable

odilitime committed:
lint: fix EmbeddingProvider already defined error (via claude)

odilitime committed:
prefer client.twitterConfig over getSetting

odilitime committed:
Merge branch 'develop' into fix/long-tweet

odilitime committed:
Merge pull request #1520 from oxSaturn/fix/long-tweet

fix: handle long tweet in utils

Titan-Node committed:
Add Livepeer model and configuration details to documentation

- Updated ModelProviderName.md to include Livepeer as a model provider.
- Added Livepeer model alias in Models.md.
- Included Livepeer gateway URL in quickstart.md and configuration.md.
- Enhanced documentation to guide users on setting up Livepeer models.

vanshika-srivastava committed:
Update package.json with build-docker command

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into fix/evm-plugin

odilitime commented:
I think there's some duplication of the work done in #1456 but I did my best to merge the best of both

odilitime committed:
remove duplicate import

odilitime committed:
remove duplicate import, remove duplicate swap call

HowieDuhzit committed:
feat: add /:agentId/speak endpoint for text-to-speech functionality

- Implemented a new POST endpoint that allows users to send text to a specified agent.
- The endpoint processes the message, generates a response, and converts it to speech using the ElevenLabs API.
- Added error handling for missing text, agent not found, and ElevenLabs API errors.
- Responses are streamed back as audio in MPEG format.

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge branch 'develop' into main

odilitime committed:
Merge pull request #1528 from HowieDuhzit/main

feat: add /:agentId/speak endpoint for text-to-speech functionality

odilitime committed:
Merge branch 'develop' into add-livepeer-image-gen-1

vanshika-srivastava committed:
Merge branch 'develop' into main

shakkernerd committed:
chore: update package.json

shakkernerd committed:
Merge pull request #1527 from vanshika-srivastava/main

Fix: Update package.json with build-docker command to match the dockerfile command

odilitime committed:
Merge branch 'develop' into add-livepeer-image-gen-1

odilitime committed:
Merge pull request #1525 from Titan-Node/add-livepeer-image-gen-1

feat: Add Livepeer Image Provider

shakkernerd committed:
Merge branch 'develop' into main

odilitime committed:
remove unused var

shakkernerd committed:
Merge pull request #839 from tsubasakong/main

feat: Add Custom System Prompt Support for plugin-image-generation

shakkernerd committed:
chore: add eslint support

shakkernerd committed:
chore: let -> const

odilitime committed:
remove unused vars per lint

shakkernerd committed:
chore: remove unused imports

odilitime committed:
Merge branch 'develop' into odi-fix

shakkernerd committed:
fix: unused variable

shakkernerd committed:
fix: unused variable

odilitime committed:
Merge pull request #1529 from elizaOS/odi-fix

chore: remove unused vars

shakkernerd committed:
fix: unused variable

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
Merge branch 'develop' into pr-842

shakkernerd committed:
Merge branch 'develop' into pr-842

shakkernerd committed:
chore: add eslint support

shakkernerd committed:
chore: let -> const

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge branch 'develop' into pr-1126

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: bump version to v.0.1.7-alpha.2

shakkernerd committed:
Merge pull request #1530 from elizaOS/new_version_0.1.7-alpha.2

chore: bump version to v.0.1.7-alpha.2

shakkernerd committed:
Merge pull request #1519 from elizaOS/develop

chore: 0.1.7-alpha.2 prep, develop => main

actions-user committed:
chore: update changelog

SK1989sL committed:
Update README.md

RedHorse823 committed:
Update README_CN - Model Configuration

The original model configuration guide is misleading, it tells readers to configure model provider by change the `XAI_MODEL` parameters.

But this is for the XAI configurations, i update the description completely with the right configuration steps.

swizzmagik committed:
Merge remote-tracking branch 'upstream/Main'

yanushevitz committed:
Added Polish Readme file

yanushevitz committed:
Fixed polish's readme typo's

yanushevitz committed:
Rename Readme_PL.md to README_PL.md

swizzmagik committed:
feat: fix chat client to autoscroll

swizzmagik committed:
chore: update changelog

mcp97 committed:
Merge pull request #1538 from swizzmagik/feat/add-autoscroll-chat-client

mcp97 committed:
Merge branch 'develop' into Polish_readme

mcp97 committed:
Merge pull request #1537 from yanushevitz/Polish_readme

Polish readme

mcp97 committed:
Merge branch 'develop' into Update-ReadMe-CN

mcp97 committed:
Merge pull request #1535 from RedHorse823/Update-ReadMe-CN

Update README_CN - Model Configuration

mcp97 committed:
Merge branch 'develop' into patch-1

mcp97 committed:
Merge pull request #1534 from SK1989sL/patch-1

docs: fix grammar/typos in README

odilitime committed:
Merge branch 'develop' into fix/evm-plugin
